regolith/tools.py

```diff
@@ -755,3 +755,34 @@ def update_schemas(default_schema, user_schema):
             updated_schema[key] = user_schema[key]
 
     return updated_schema
+
+
+def group(db, by):
+    """
+    Group the document in the database according to the value of the doc[by] in db.
+
+    Parameters
+    ----------
+    db : generator
+        The database of documents.
+    by : basestring
+        The key to group the documents.
+
+    Returns
+    -------
+    grouped: dict
+        A dictionary mapping the feature value of group to the list of docs. All docs in the same generator have
+        the same value of doc[by].
+    """
+    grouped = {}
+    doc: dict
```

Comment:
what is this?

Reply:
It is the annotation in pyhton >= 3.6 It does nothing but tell the developer that the variable 'doc' is a dictionary. By using the annotation, there are two advantages using it: (1) pycharm and other IDE can autocomplete the method for the variable and check if it is used correctly (2) other developer can know what they can do with the doc when they are trying to develop this function further.

Reply:
Sorry, the terminology for this should be "type hints" not "annotation".

Reply:
I see.  Is regolith already py >= 3.6 in the requirements?

Reply:
Yes. In the travis ci. The python is 3.6. In the setup tools, it is "Programming Language :: Python :: 3" so it will install the newest python. Either way, the python has the type hints.